tracer/src/Datadog.Trace/Sampling/SpanSampler.cs

```diff
@@ -0,0 +1,69 @@
+// <copyright file="SpanSampler.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.Sampling;
+
+/// <summary>
+///     Represents a sampler for single span ingestion.
+/// </summary>
+internal class SpanSampler : ISpanSampler
+{
+    private static readonly string SamplingMechanismString = SamplingMechanism.SpanSamplingRule.ToString();
+
+    private readonly List<ISpanSamplingRule> _rules = new List<ISpanSamplingRule>();
+
+    public SpanSampler(IEnumerable<ISpanSamplingRule> rules)
+    {
+        if (rules is null)
+        {
+            throw new ArgumentNullException(nameof(rules));
+        }
+
+        _rules = rules.ToList();
+    }
```

Comment:
Another option to consider - if you call `rules.ToList()` where this is created (in `TracerManager`) you could skip creating the `SpanSampler` entirely if there are no rules, which would then skip calling `MakeSamplingDecision` etc with (tiny) perf benefits? Maybe not worth the effort, but also not much reason not to do it I think 🙂

Reply:
I made it to match the `TraceSampler` where the `SpanSampler` will get initialized when there are no rules and just return false whenever `MakeSamplingDecision` is made. 

But sounds like maybe coming back to this and looking at how both the `TraceSampler`/`SpanSampler` are created in `TracerManager` would be beneficial to make them consistent and potentially both get that performance benefit